### PR TITLE
#166272054 setup webpack

### PR DESCRIPTION
**Pull request by EmmanuelSage**
#### What does this PR do?

- This PR sets up webpack
#### Description of task to be completed?
- install dependencies
- setup eslint
- setup postcss
- setup babel
- setup webpack configuration
- test with basic render

#### How should this be manually tested?
  - $ git clone https://github.com/andela/king-kong-ah-frontend.git
  - $ npm install
  - $ npm start

#### Any background context you want to provide?
- npm must have been installed to run

#### What are the relevant pivotal tracker stories?
- PT Story link - [#166272054](https://www.pivotaltracker.com/story/show/166272054)

#### Screenshots (if appropriate)
Screenshot 2019-07-02 at 12 22 07 PM

#### Questions:
- N/A

### CODE REVIEW

**Review by mosinmiloluwa01 — APPROVED** [review]

**Review by Eazybee — APPROVED** [review]

File: .postcssrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+ "postcss": {
```
Comment (benfluleck) [inline .postcssrc]: What is this doing here ?

File: README.md [hunk 2]
```diff
@@ -41,4 +41,4 @@ npm test
 - Ugege Lewis
 - Soyoye Olusoji
 - Akinyele Oluwatosin
-- Owoso Mosinmiloluwa
\ No newline at end of file
+- Owoso Mosinmiloluwa
```
Comment (benfluleck) [inline README.md:45]: TTL nko 😢

File: package.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    "eslint-config-airbnb": "^17.1.0",
+    "eslint-plugin-import": "^2.16.0",
+    "eslint-plugin-jsx-a11y": "^6.2.1",
+    "eslint-plugin-react": "^7.12.4",
+    "file-loader": "^3.0.1",
+    "html-webpack-plugin": "^3.2.0",
+    "husky": "^3.0.0",
+    "lint-staged": "^9.0.0",
+    "mini-css-extract-plugin": "^0.5.0",
+    "node-sass": "^4.12.0",
```
Comment (benfluleck) [inline package.json]: You are not planning to use sass are you?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    "postcss-loader": "^3.0.0",
+    "sass-loader": "^7.1.0",
```
Comment (benfluleck) [inline package.json]: Why is this here?

File: webpack/webpack.common.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    rules: [
+      {
+        test: /\.(jsx|js)$/,
+        exclude: /node_modules/,
+        use: [
+          {
+            loader: 'babel-loader',
+            options: {
+              presets: ['@babel/preset-env', '@babel/preset-react'],
+              plugins: [
```
Comment (benfluleck) [inline webpack/webpack.common.js:26]: Why is your babel config different from your babel config in webpack

File: webpack/webpack.common.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                '@babel/plugin-proposal-class-properties',
+                '@babel/plugin-transform-runtime',
+                '@babel/plugin-transform-arrow-functions',
+              ],
+            },
+          },
+        ],
+      },
+      {
+        test: /\.(css|scss|sass)$/,
```
Comment (benfluleck) [inline webpack/webpack.common.js]: Where is sass coming from na

File: webpack/webpack.development.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  devServer: {
+    compress: true,
```
Comment (benfluleck) [inline webpack/webpack.development.js]: I do not see the benefits of this line

File: webpack/webpack.production.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  performance: {
+    hints: false,
+  },
+  optimization: {
+    minimizer: [uglifyJS],
+    splitChunks: {
+      cacheGroups: {
+        styles: {
+          name: 'styles',
+          test: /\.scss$/,
```
Comment (benfluleck) [inline webpack/webpack.production.js]: Once again I see sass in your project 🤔

**Review by benfluleck — APPROVED** [review]
